add post announcing compiler team re-org (#1421)
rust-lang/blog.rust-lang.org: posts/inside-rust/2024-10-31-compiler-team-reorg.md

---
layout: post
title: Re-organising the compiler team and recognising our team members
author: davidtwco and wesleywiser
team: the compiler team <https://www.rust-lang.org/governance/teams/compiler>
---
Back in June, the compiler team merged [RFC 3599][rfc] which re-structured the team to ensure the

> team's policies and processes can support the maintainance of the Rust compiler going forward.

Reviewer: ```suggestion
team's policies and processes can support the maintenance of the Rust compiler going forward.
```

    - apiraino is an invaluable member of the compiler team, handily tackling the team's
      operational work (agenda preparation, meeting notes, automation, etc) so that the team's work
      progresses smoothly each week.
- [b-naber](https://github.com/b-naber), team member
    - b-naber has been contributing to the compiler since 2020 and has worked on the compiler's
      constant evaluation, on constant generics and on generic associated types in that time.
- [bjorn3](https://github.com/bjorn3), team member
    - bjorn3 has been an active contributor to the compiler since 2017, becoming a compiler team
      contributor in 2020. bjorn has been instrumental in the compiler's support for multiple
      codegen backends and is the primary author of the Cranelift codegen backend.
- [BoxyUwU](https://github.com/BoxyUwU), maintainer
    - BoxyUwU has been contributing relentlessly since 2020, immediately jumping into some of the
      more challenging and technical areas in the compiler. Boxy is a member of the types team and
      has contributed heavily to the implementation and design of constant generics.
- [camelid](https://github.com/camelid), team member

>     - camelid has contributed for many years, improving project documentation, compiler diagnostics
>       and making many refactorings and cleanups on the compiler codebase. 

Reviewer 2: ```suggestion
    - camelid has contributed for many years, improving constant generics, project documentation, and compiler diagnostics
      and making many refactorings and cleanups on the compiler codebase. 
```

      in the language, RalfJ is the team's foremost expert on the operational semantics of the
      language and also deeply involved in const evaluation. He has been working on achieving sound
      semantics down to lowest levels of the compiler.
- [saethlin](https://github.com/saethlin), maintainer
    - saethlin has made significant improvements to MIR, Miri, and codegen in their contributions
      since starting in 2021 and has become an invaluable source of knowledge on the language's
      operational semantics/unsafe code guidelines.
- [scottmcm](https://github.com/scottmcm), team member
    - scottmcm is a member of the language team who also regularly implements improvements in the
      compiler, particularly in the MIR and compiler's codegen, always trying to get to the perfect
      machine code.
- [SparrowLii](https://github.com/sparrowlii), maintainer
    - SparrowLii is a relatively new member of the team who has resurrected and led the effort to
      parallelize the compiler and has been responsible for the great strides that effort has made
      alongside members of the parallel rustc and compiler performance working groups.

> - [spastorino](https://github.com/spastorino), maintainer
>     - spastorino is another alum of the non-lexical lifetimes working group, starting to contribute
>       in late 2017. Since NLL, spastorino has implemented negative impls in coherence, refactored
>       return position impl trait in trait code to lower as a GAT, and has done a lot of bugfixes
>       to the compiler in general and with to the type system as a member of the types team.

Reviewer 3: ```suggestion
- [spastorino](https://github.com/spastorino), maintainer
    - spastorino is another alum of the non-lexical lifetimes working group, starting to contribute
      in late 2017. Since NLL, spastorino has implemented negative impls in coherence, refactored
      return position impl trait in trait code to lower as a GAT, and has done a lot of refactors/bugfixes
      to the compiler.
```
